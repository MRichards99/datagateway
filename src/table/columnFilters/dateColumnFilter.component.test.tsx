import React from 'react';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import DateColumnFilter from './dateColumnFilter.component';

describe('Text filter component', () => {
  let shallow;
  let mount;

  beforeEach(() => {
    shallow = createShallow();
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <DateColumnFilter label="test" onChange={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('calls the onChange method correctly when filling out startDate then endDate, then clearing startDate then endDate', () => {
    const onChange = jest.fn();

    const wrapper = mount(
      <DateColumnFilter label="test" onChange={onChange} />
    );

    const startDateFilterInput = wrapper.find('input').first();
    startDateFilterInput.instance().value = '2019-08-06';
    startDateFilterInput.simulate('change');

    expect(onChange).toHaveBeenLastCalledWith({
      startDate: '2019-08-06',
    });

    const endDateFilterInput = wrapper.find('input').last();
    endDateFilterInput.instance().value = '2019-08-06';
    endDateFilterInput.simulate('change');

    expect(onChange).toHaveBeenLastCalledWith({
      startDate: '2019-08-06',
      endDate: '2019-08-06',
    });

    startDateFilterInput.instance().value = '';
    startDateFilterInput.simulate('change');

    expect(onChange).toHaveBeenLastCalledWith({
      endDate: '2019-08-06',
    });

    endDateFilterInput.instance().value = '';
    endDateFilterInput.simulate('change');

    expect(onChange).toHaveBeenLastCalledWith(null);
  });

  it('calls the onChange method correctly when filling out endDate then startDate, then clearing endDate then startDate', () => {
    const onChange = jest.fn();

    const wrapper = mount(
      <DateColumnFilter label="test" onChange={onChange} />
    );

    const endDateFilterInput = wrapper.find('input').last();
    endDateFilterInput.instance().value = '2019-08-06';
    endDateFilterInput.simulate('change');

    expect(onChange).toHaveBeenLastCalledWith({
      endDate: '2019-08-06',
    });

    const startDateFilterInput = wrapper.find('input').first();
    startDateFilterInput.instance().value = '2019-08-06';
    startDateFilterInput.simulate('change');

    expect(onChange).toHaveBeenLastCalledWith({
      startDate: '2019-08-06',
      endDate: '2019-08-06',
    });

    endDateFilterInput.instance().value = '';
    endDateFilterInput.simulate('change');

    expect(onChange).toHaveBeenLastCalledWith({
      startDate: '2019-08-06',
    });

    startDateFilterInput.instance().value = '';
    startDateFilterInput.simulate('change');

    expect(onChange).toHaveBeenLastCalledWith(null);
  });
});
