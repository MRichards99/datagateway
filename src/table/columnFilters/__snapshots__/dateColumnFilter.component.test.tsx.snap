// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text filter component renders correctly 1`] = `
<form>
  <MuiPickersUtilsProvider
    utils={[Function]}
  >
    <WrappedKeyboardPicker
      KeyboardButtonProps={
        Object {
          "size": "small",
        }
      }
      allowKeyboardControl={true}
      clearable={true}
      format="yyyy-MM-dd"
      inputProps={
        Object {
          "aria-label": "test date filter from",
        }
      }
      invalidDateMessage="Invalid Date Format"
      maxDate={2100-01-01T00:00:00.000Z}
      maxDateMessage="Date should not be after maximal date"
      minDate={1900-01-01T00:00:00.000Z}
      minDateMessage="Date should not be before minimal date"
      onChange={[Function]}
      openTo="date"
      placeholder="From..."
      value={null}
      views={
        Array [
          "year",
          "date",
        ]
      }
    />
    <WrappedKeyboardPicker
      KeyboardButtonProps={
        Object {
          "size": "small",
        }
      }
      allowKeyboardControl={true}
      clearable={true}
      format="yyyy-MM-dd"
      inputProps={
        Object {
          "aria-label": "test date filter to",
        }
      }
      invalidDateMessage="Invalid Date Format"
      maxDate={2100-01-01T00:00:00.000Z}
      maxDateMessage="Date should not be after maximal date"
      minDate={1900-01-01T00:00:00.000Z}
      minDateMessage="Date should not be before minimal date"
      onChange={[Function]}
      openTo="date"
      placeholder="To..."
      value={null}
      views={
        Array [
          "year",
          "date",
        ]
      }
    />
  </MuiPickersUtilsProvider>
</form>
`;
